### PR TITLE
Publish the public API as a JSON object

### PR DESCRIPTION
**Pull request by zeke**
This is a followup to https://github.com/electron/electron/pull/6408, a closed PR which I was unable to re-open after a force-push. Please see that PR for context on this effort.

I finished everything in the [electron-docs-linter milestone](https://github.com/zeke/electron-docs-linter/milestone/1) and with @kevinsawicki's help I've added the necessary code to upload the generated JSON object along with the Electron release.

@zcbenz this is ready for review.

### CODE REVIEW

File: script/create-dist.py [hunk 1]
```diff
@@ -128,6 +129,8 @@ def copy_license():
                DIST_DIR)
   shutil.copy2(os.path.join(SOURCE_ROOT, 'LICENSE'), DIST_DIR)
 
+def copy_api_json():
+    shutil.copy2(os.path.join(OUT_DIR, 'electron-api.json'), DIST_DIR)
```
Comment (kevinsawicki) [inline script/create-dist.py]: This should be a 2 space indent.

File: script/create-dist.py [hunk 2]
```diff
@@ -85,6 +85,7 @@ def main():
   copy_chrome_binary('chromedriver')
   copy_chrome_binary('mksnapshot')
   copy_license()
+  copy_api_json()
```
Comment (MarshallOfSound) [inline script/create-dist.py]: Semantically this doesn't read very well.

Perhaps `copy_generated_json_api()`?

To be honest I can't even come up with a nice reading function name 😢

Comment (enlight) [inline script/create-dist.py]: Other suggestions: `copy_json_api` or `copy_json_api_file`

**Comment by zeke** [comment]
Travis CI is failing because it's using node 0.10

AppVeyor is failing on something that I think is unrelated:

> ENOENT: no such file or directory, open 'C:\projects\electron\spec\does-not-exist\manifest.json

`electron-osx-x64` CI is failing in the `npm run lint` task, but not failing in my local macOS environment.

@kevinsawicki @zcbenz any tips on how I can isolate this new stuff to only run in the necessary CI environment(s)?

File: script/upload.py [hunk 3]
```diff
@@ -78,6 +78,7 @@ def main():
   upload_electron(github, release, os.path.join(DIST_DIR, DIST_NAME))
   upload_electron(github, release, os.path.join(DIST_DIR, SYMBOLS_NAME))
   if PLATFORM == 'darwin':
+    upload_electron(github, release, os.path.join(DIST_DIR, 'electron-api.json'))
```
Comment (kevinsawicki) [inline script/upload.py]: This line is over 80 characters and causing `script/pylint.py` to fail.

Comment (zeke) [inline script/upload.py]: Thanks. Fixed.

File: script/upload.py [hunk 4]
```diff
@@ -78,6 +78,8 @@ def main():
   upload_electron(github, release, os.path.join(DIST_DIR, DIST_NAME))
   upload_electron(github, release, os.path.join(DIST_DIR, SYMBOLS_NAME))
   if PLATFORM == 'darwin':
+      upload_electron(github, release, os.path.join(DIST_DIR,
```
Comment (zcbenz) [inline script/upload.py]: The indention is not correct.

**Comment by zeke** [comment]
All CI builds are passing but Travis, which is giving this error:

```
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Traceback (most recent call last):
  File "./script/cibuild", line 120, in <module>
    sys.exit(main())
```

I tried a manual rebuild but it seems to be failing in the same spot.

**Comment by MarshallOfSound** [comment]
@zeke It looked like travis was having some issues this morning (other builds have been failing with this issue as well)

**Comment by zcbenz** [comment]
This looks good to me, feel free to merge when you think it is ready.

**Pull request merged**